apps/back/src/DataFixtures/UserFixtures.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace App\DataFixtures;
+
+use App\Enum\UserRole;
+use App\Service\Factory\UserFactory;
+use DateTimeImmutable;
+use Doctrine\Bundle\FixturesBundle\Fixture;
+use Doctrine\Persistence\ObjectManager;
+
+class UserFixtures extends Fixture
+{
+    const IS_ADMIN = 'isAdmin';
+    const VIEWS_COUNT = 'viewsCount';
+    const LAST_COURSE_VIEW_DATE = 'lastCourseViewDate';
+
+    public function __construct(private readonly UserFactory $userFactory)
+    {
+    }
+
+    public function load(ObjectManager $manager): void
+    {
+        /* Get necessary data to create users (ex: is admin) */
+        $courseUserData = self::getCourseUserData();
+        $i = 1;
+
+        /* Foreach user, create user data and persist it */
+        foreach ($courseUserData as $userData) {
```

Comment:
Rien de bien transcendant mais pour ne pas avoir à maintenir une variable `$i`. tu peux utiliser la clé de ta liste : 
```suggestion
        /* Foreach user, create user data and persist it */
        foreach ($courseUserData as $i => $userData) {
```
Faut juste faire une +1 pour l'utiliser ensuite. C'est comme tu le sens